Source/cairo/WaylandCairoSurface.m

```diff
@@ -176,62 +184,41 @@ - (void) dealloc
 
 - (NSSize) size
 {
-    NSDebugLog(@"WaylandCairoSurface: size");
+//    NSDebugLog(@"WaylandCairoSurface: size");
     struct window *window = (struct window*) gsDevice;
     return NSMakeSize(window->width, window->height);
 }
 
 - (void) setSurface: (cairo_surface_t*)surface
 {
-    NSDebugLog(@"WaylandCairoSurface: setSurface");
+//    NSDebugLog(@"WaylandCairoSurface: setSurface");
     _surface = surface;
 }
 
 - (void) handleExposeRect: (NSRect)rect
 {
-    NSDebugLog(@"handleExposeRect");
     struct window *window = (struct window*) gsDevice;
-    cairo_surface_t *cairo_surface = _surface;
-    double  backupOffsetX = 0;
-    double  backupOffsetY = 0;
+    NSDebugLog(@"[CairoSurface handleExposeRect] %d", window->window_id);
+
     int x = NSMinX(rect);
     int y = NSMinY(rect);
     int width = NSWidth(rect);
     int height = NSHeight(rect);
 
-    NSDebugLog(@"updating region: %dx%d %dx%d", x, y, width, height);
-
-    if (cairo_surface_status(cairo_surface) != CAIRO_STATUS_SUCCESS)
-    {
-	NSWarnMLog(@"cairo initial window error status: %s\n",
-		   cairo_status_to_string(cairo_surface_status(_surface)));
+    window->buffer_needs_attach = YES;
+    if (window->configured) {
+        window->buffer_needs_attach = NO;
+        wl_surface_attach(window->surface, window->buffer, 0, 0);
```

Comment:
Do we need the attach call here even when it was already called in -initWithDevice:?

I would expect that we have a flag that tells us, whether this was already called if there is no consistent order in which these methods get called.

Reply:
it might be superfluous this code, i need to debug it and add a comment for this edge case, the flag used elsewhere is `window->buffer_needs_attach`

Reply:
@fredkiefer i added a comment in the code to explain better when this is needed:
usually the attach() is handled in the configure callback, when resizing a window already configured then the new buffer created should be attached immediately. maybe we could move the attach outside of the surface class...